src/components/KYCWall/types.ts

```diff
@@ -0,0 +1,86 @@
+import CONST from '@src/CONST';
+import {BankAccountList, FundList, ReimbursementAccount, Report, UserWallet, WalletTerms} from '@src/types/onyx';
+import { Route } from '@src/ROUTES';
+import { ValueOf } from 'type-fest';
+import { ForwardedRef, SyntheticEvent } from 'react';
+import { NativeTouchEvent } from 'react-native';
+
+type Source = ValueOf<typeof CONST.KYC_WALL_SOURCE>
+
+type WalletTermsWithSource = WalletTerms & {source: Source}
+
+type TransferMethod = ValueOf<typeof CONST.WALLET.TRANSFER_METHOD_TYPE>
+
+type KYCWallProps = {
+    /** Route for the Add Bank Account screen for a given navigation stack */
+    addBankAccountRoute: Route;
+
+    /** Route for the Add Debit Card screen for a given navigation stack */
+    addDebitCardRoute?: Route;
+
+    /** Route for the KYC enable payments screen for a given navigation stack */
+    enablePaymentsRoute: Route;
+
+    /** Listen for window resize event on web and desktop */
+    shouldListenForResize?: boolean;
+
+    /** Wrapped components should be disabled, and not in spinner/loading state */
+    isDisabled?: boolean;
+
+    /** The user's wallet */
+    userWallet?: UserWallet;
+
+    /** Information related to the last step of the wallet activation flow */
+    walletTerms?: WalletTermsWithSource;
+
+    /** The source that triggered the KYC wall */
+    source: Source;
+
+    /** When the button is opened via an IOU, ID for the chatReport that the IOU is linked to */
+    chatReportID?: string;
+
+    /** List of user's cards */
+    fundList?: FundList;
+
+    /** List of bank accounts */
+    bankAccountList?: BankAccountList;
+
+    /** The chat report this report is linked to */
+    chatReport?: Report;
+
+    /** The IOU/Expense report we are paying */
+    iouReport: Report;
+
+    /** The reimbursement account linked to the Workspace */
+    reimbursementAccount?: ReimbursementAccount;
```

Comment:
I think you describe Onyx Props [there](https://github.com/Expensify/App/pull/33714/files#diff-752b992e7933236d8b651b3b2778993788123d5cb84debc2adabc0bbe7df547fR25), so no need to put it in this type as well.